music-player/src/components/AddSongToPlaylist.tsx

```diff
@@ -0,0 +1,80 @@
+import { Modal, Button } from 'antd';
+import * as React from 'react';
+import { connect } from 'react-redux';
+import { Dispatch } from 'redux';
+
+import { State } from '../types';
+import { getPlaylists } from '../reducers';
+import { getAddSongToPlaylistAction } from '../actions';
+
+class AddSongToPlaylist extends React.Component<
+  {
+    playlistsNotIncludeSongIndex: string[];
+    realIndex: number;
+    dispatch: Dispatch;
+  },
+  { show: boolean }
+> {
+  constructor(props: any) {
+    super(props);
+    this.state = { show: false };
+  }
+
+  closeWindowFunction = () => this.setState({ show: false });
+
+  render = () => {
+    const { playlistsNotIncludeSongIndex, dispatch } = this.props;
+    return (
+      <span>
+        <Button
+          className="song-button add-song-button"
```

Comment:
Again with styled components.
(I'm not sure I actually want you to fix that or if it's a waste of time)